R/model-tree.R

```diff
@@ -681,6 +733,37 @@ color_tree_by <- function(tree_data, color_by = "run"){
   return(tree_data)
 }
 
+#' Create a size column based on the unique _numeric_ (or integer) values of
+#' another column.
+#' @inheritParams make_tree_tooltip
+#' @inheritParams model_tree
+#' @noRd
+size_tree_by <- function(tree_data, size_by = NULL){
+  if(!is.null(size_by)){
+    checkmate::assert_true(size_by %in% names(tree_data))
+    tree_data$node_size <- tree_data[[size_by]]
+    # Scale size with numeric value
+    if(inherits(tree_data$node_size, c("numeric", "integer"))){
+      # Set node sizes with NA values (including start node) to mean value
+      mean_val <- mean(tree_data$node_size, na.rm = TRUE)
+      tree_data$node_size[is.na(tree_data$node_size)] <- mean_val
+    }else{
+      col_class <- class(tree_data[[size_by]])
+      cli::cli_warn(
+        c(
+          "Only numeric columns are supported. Column {.val {size_by}} is {.val {col_class}}",
+          "i" = "Setting node size to constant"
+        )
+      )
+      tree_data$node_size <- 1
```

Comment:
Think it's fine to leave, but what's your thinking behind setting it to a constant size rather than warning and leaving the default "leafCount" behavior?

Reply:
Well im using `size_tree_by` for some error handling, so id have to return something to indicate a change to make it "leafCount" (probably not the `node_size` being 1 since that feels sloppy and prone to unintended consequences).  Im am open to making that the default if you think that's preferable

Reply:
I don't feel strongly about what the fallback is given it's for a mis-specified case.  It does seem odd to me to let how you structured an internal helper be the reason for why you should set it to a constant rather than leave it at a default, but there are more important things to focus on with this change.

Reply:
I addressed this and your next comment in [e2c0909](https://github.com/metrumresearchgroup/bbr/pull/720/commits/e2c0909c88736176ccee06a70505f66231115e22). Let me know what you think about that handling.

Reply:
I think the behavior change is good.  My initial reaction is that I don't find sneaking it through an attribute easy to follow, but I think it's fine.  Let's leave this be.

Reply:
Edit: I should add that the default of _this function_ is leafCount, though the default of `collapsibleTreeNetwork` is the handling I had before. From the docs:

> numeric column that will be used to determine relative node size. Default is to have a constant node size throughout. 'leafCount' can also be used here (cumulative count of a node's children), or 'count' (count of node's immediate children).

I still think the change I made makes sense given the docs for `model_tree`. For our use case I think `leafCount` makes sense as the default conceptually too.

Reply:
> I should add that the default of _this function_ is leafCount, though the default of `collapsibleTreeNetwork` is the handling I had before.

Hmm, I having a hard time figuring how that's relevant if it's never been exposed to model_tree users.  Perhaps I'm missing something.

To me, the point is just simply to align the fallback with the default `model_tree` behavior.  (Whether leafCount is a good default is of course another question, but that's already been decided before this PR.)

Reply:
>having a hard time figuring how that's relevant if it's never been exposed to model_tree users

Not relevant to users, just thought you may want to know that. I don't believe that's why I set `tree_data$node_size <- 1` previously (think I just wanted to make it obvious an incorrect `size_by` column was specified so they'd correct their code, as well as keeping the returned data frame consistent) but just noticed that now when looking through the docs.